FSharpLu.Json/Compact.fs

```diff
@@ -10,18 +13,33 @@ open Microsoft.FSharp.Reflection
 type CompactUnionJsonConverter() =
     inherit Newtonsoft.Json.JsonConverter()
 
+    let processWithContractResolver (serializer:JsonSerializer) (name:string) =
```

Comment:
I am wondering whether the dynamic match on `ContractResolver` might impact performance. It may be worth inlining this function. Something like
```
let inline processWithContractResolver (serializer:JsonSerializer) =
   let toPascal (name:string) =
            if System.Char.IsUpper (name, 0) then name
            else string(System.Char.ToUpper name.[0]) + name.Substring(1)  
        let toCamel (name:string) =
            if System.Char.IsLower (name, 0) then name
            else string(System.Char.ToLower name.[0]) + name.Substring(1)    
        match serializer.ContractResolver with 
        | :? CamelCasePropertyNamesContractResolver -> toCamel
        | :? DefaultContractResolver -> toPascal
        | _ -> id
```